File: Frontend/src/api/battle.js
import { api } from './client';

export function startBattle({ questId, excuseText }) {
  return api.post('/battle/start', {
    quest_id: questId,
    excuse_text: excuseText,
  });
}

export function submitTurn({ battleId, userInput }) {
  return api.post(`/battle/${battleId}/turn`, {
    user_input: userInput,
  });
}

File: backend/app/schemas/battle.py
from pydantic import BaseModel
from typing import Optional
from uuid import UUID


# ── POST /api/battle/start ─────────────────────────────
class BattleStartRequest(BaseModel):
    quest_id: str   # "promise" / "homework" / "social" / "work" / "family"
    excuse_text: str


class BattleStartResponse(BaseModel):
    battle_id: str
    suspicion: int      # 0~100
    enemy_line: str


# ── POST /api/battle/{battle_id}/turn ─────────────────
class BattleTurnRequest(BaseModel):
    user_input: str


class DefenseType(BaseModel):
    defense_type: str
    reason: str


class BattleTurnResponse(BaseModel):
    suspicion: int
    enemy_reaction: str
    enemy_line: Optional[str]
    is_final: bool
    success: Optional[bool]
    defense_type: Optional[DefenseType]
